libs/platform/src/lib/components/dynamic-page/dynamic-page-header/header/dynamic-page-header.component.ts

```diff
@@ -0,0 +1,424 @@
+     */
+    private _toggleSubscription: Subscription;
+
+    /**
+     * @hidden
+     * subscription for when expand header is called
+     */
+    private _expandSubscription: Subscription;
+
+    /**
+     * @hidden
+     * subscription for when collapse header is called
+     */
+    private _collapseSubscription: Subscription;
+
+    /** @hidden */
+    constructor(
+        private _cd: ChangeDetectorRef,
+        private _elementRef: ElementRef<HTMLElement>,
+        private _renderer: Renderer2,
+        protected _dynamicPageConfig: DynamicPageConfig,
+        private _dynamicPageService: DynamicPageService
+    ) {
+        if (this.collapsible) {
+            this._toggleSubscription = this._dynamicPageService.$toggle.subscribe(() => {
+                this.toggleCollapse();
+            });
+            this._expandSubscription = this._dynamicPageService.$expand.subscribe(() => {
+                this.collapseHeader(false);
+            });
+            this._collapseSubscription = this._dynamicPageService.$collapse.subscribe(() => {
+                this.collapseHeader(true);
+            });
+        }
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        if (this._isCollapsibleCollapsed()) {
+            this._setStyleToHostElement('z-index', 1);
+        }
+        this._calculateExpandCollapseAriaLabel();
+        this._calculatePinUnpinAriaLabel();
+    }
+
+    /** @hidden */
+    ngAfterViewInit(): void {
+        if (this._background) {
+            this._setBackgroundStyles(this._background);
+        }
+        if (this.size) {
+            this._setSize(this.size);
+        }
+    }
+
+    /**@hidden */
+    ngOnDestroy(): void {
+        this._toggleSubscription.unsubscribe();
+        this._expandSubscription.unsubscribe();
+        this._collapseSubscription.unsubscribe();
+    }
+
+    collapseHeader(val: boolean): any {
+        if (this._isPinned()) {
+            return;
+        }
+        if (this.collapsed !== val) {
+            this.collapsed = val;
+            this.expandCollapseActions();
+        }
+    }
+
+    /** Handles expanded/collapsed event */
+    toggleCollapse(): void {
+        if (this._isPinned()) {
+            return;
+        }
+
+        this.collapsed = !this._dynamicPageService.getIsCollapsed();
+        this.expandCollapseActions();
+    }
+
+    /**
+     * click action on pin button
+     */
+    onPinned(): void {
+        this.pinned = !this.pinned;
+        if (this.pinned) {
+            this._collapsible = false;
+        } else {
+            this._collapsible = this.collapsible; // reset
+        }
+        this._calculatePinUnpinAriaLabel();
+    }
+
+    /**
+     * return the element reference.
+     */
+    elementRef(): ElementRef<any> {
+        return this._elementRef;
+    }
+
+    /**
+     * handles actions like style changes and emit methods on expand/collapse
+     */
+    private expandCollapseActions(): void {
```

Comment:
CAn you add `_` prefix there?

Reply:
oops, missed it. Thanks for noticing, added now.